docs/ingestion-beam/ecosystem-decryptor-job.md

```diff
@@ -0,0 +1,26 @@
+# EcosystemDecryptor Job
+
+A job for decrypting identifiers used in Account Ecosystem Telemetry.
+Defined in the `com.mozilla.telemetry.EcosystemDecryptor` class ([source](https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-beam/src/main/java/com/mozilla/telemetry/EcosystemDecryptor.java)).
+On startup, this job loads a bundle of private keys by contacting GCP's KMS service.
+
+## Transforms
+
+These transforms are currently executed against each message in order.
+
+### Decrypt Account Ecosystem Telemetry Identifiers
+
+Strips encrypted AET attributes from the message, decrypts them using the
+loaded bundle of data pipeline private keys, and adds attributes for the
+decrypted IDs:
+
+* Input attribute `x_ecosystem_anon_id` will be transformed to output attribute
+  `ecosystem_user_id`
+* Input attribute `x_prev_ecosystem_anon_id` (if present) will be transformed to
+  output attribute `prev_ecosystem_user_id`
+
+## Formats
+
+The input `x_ecosystem_anon_id` and `x_prev_ecosystem_anon_id` attribute strings must
+be JOSE JWE objects in Compact Serialization form. The emitted `ecosystem_user_id`
+and `prev_ecosystem_user_id` attribute strings are UUIDs in 36-character hex standard form.
```

Comment:
Is it important that the `ecosystem_user_id` strictly be a UUID?  We can guarantee the "alphabet" (hexadecimal) and its uniqueness, but I don't know the format can be strictly guaranteed.   UUID carries some restrictions that are kind of inconvenient.

The Firefox Account `uid` is a 32-character string of hexadecimal, but is not strictly a UUID.  I think it safer to expect the `ecosystem_user_id` match an identical format.

Reply:
UUID is not a hard requirement; I'm mostly concerned with having the format be predictable and well-documented. A 32-character hex string sounds reasonable.